packages/apps/reputation-oracle/server/src/modules/slack/slack.service.ts

```diff
@@ -0,0 +1,169 @@
+      throw error;
+    }
+  }
+
+  private async openModal(triggerId: string, modalView: any): Promise<void> {
+    try {
+      const response = await firstValueFrom(
+        this.httpService.post(
+          'https://slack.com/api/views.open',
+          {
+            trigger_id: triggerId,
+            view: modalView,
+          },
+          {
+            headers: {
+              Authorization: `Bearer ${this.slackConfigService.oauthToken}`,
+              'Content-Type': 'application/json',
+            },
+          },
+        ),
+      );
+
+      if (!response.data.ok) {
+        this.logger.error('Error opening Slack modal:', response.data);
+        throw new Error('Failed to open Slack modal');
+      }
+    } catch (error) {
+      this.logger.error('Error opening Slack modal:', error);
+      throw error;
+    }
+  }
+
+  async updateMessage(responseUrl: string, text: string): Promise<void> {
+    try {
+      await firstValueFrom(this.httpService.post(responseUrl, { text }));
+    } catch (error) {
+      this.logger.error('Error updating Slack message:', error);
+      throw error;
+    }
+  }
+
+  async sendAbuseNotification(data: {
+    abuseId: number;
+    chainId: ChainId;
+    escrowAddress: string;
+  }): Promise<void> {
+    const escrow = await EscrowUtils.getEscrow(
+      data.chainId,
+      data.escrowAddress,
+    );
+
+    const message = {
+      text: 'New abuse report received!',
+      attachments: [
+        {
+          title: 'Escrow',
+          fields: [
+            { title: 'Address', value: data.escrowAddress },
+            { title: 'ChainId', value: data.chainId },
+            { title: 'Manifest', value: escrow.manifestUrl },
+          ],
+        },
+        {
+          fallback: 'Actions',
+          title: 'Actions',
+          callback_id: data.abuseId,
+          color: '#3AA3E3',
+          attachment_type: 'default',
+          actions: [
+            {
+              name: 'accept',
+              text: 'Slash',
+              type: 'button',
+              style: 'primary',
+              value: AbuseDecision.ACCEPTED,
+            },
+            {
+              name: 'reject',
+              text: 'Reject',
+              type: 'button',
+              style: 'danger',
+              value: AbuseDecision.REJECTED,
+              confirm: {
+                title: 'Cancel abuse',
+                text: `Are you sure you want to cancel slash for escrow ${data.escrowAddress}?`,
+                ok_text: 'Yes',
+                dismiss_text: 'No',
+              },
+            },
+          ],
+        },
+      ],
+    };
+
+    await this.sendNotification(message);
+  }
+
+  async sendAbuseReportModal(data: {
+    abuseId: number;
+    chainId: ChainId;
+    escrowAddress: string;
+    triggerId: string;
+    responseUrl: string;
+  }) {
+    const escrow = await EscrowUtils.getEscrow(
+      data.chainId,
+      data.escrowAddress,
+    );
+    const maxAmount = (
```

Comment:
Same as for `sendAbuseNotification`: all inputs should be provided from the outside.
With current implementation when e.g. writing unit tests or playing around with this class you will have to mock `EscrowUtils` somehow, but if you just pass some inputs - you can easily test sending flow